src/box/lua/iproto.c

```diff
@@ -28,240 +28,121 @@
 
 struct mh_strnu32_t *iproto_key_translation;
 
-/**
- * IPROTO constant from `src/box/iproto_{constants, features}.h`.
- */
-struct iproto_constant {
-	/**
-	 * Constant literal, name of constant.
-	 */
-	const char *const name;
-	/**
-	 * Constant literal, value of constant.
-	 */
-	const lua_Integer val;
-};
-
 /**
  * Pushes an array of IPROTO constants onto Lua stack.
  */
 static void
-push_iproto_constant_subnamespace(struct lua_State *L, const char *subnamespace,
-				  const struct iproto_constant *constants,
-				  int constants_len)
+push_iproto_constant_subnamespace(
+	struct lua_State *L, const char *subnamespace,
+	const struct iproto_constant *constants, int constants_len,
+	const char *strip_prefix)
 {
+	size_t prefix_len = 0;
+	if (strip_prefix != NULL)
+		prefix_len = strlen(strip_prefix);
 	lua_createtable(L, 0, constants_len);
 	for (int i = 0; i < constants_len; ++i) {
-		lua_pushinteger(L, constants[i].val);
-		lua_setfield(L, -2, constants[i].name);
+		const char *name = constants[i].name;
+		int value = constants[i].value;
+		if (strstr(name, "RESERVED"))
+			continue;
+		assert(prefix_len == 0 ||
+		       strncmp(name, strip_prefix, prefix_len) == 0);
+		lua_pushinteger(L, value);
+		lua_setfield(L, -2, name + prefix_len);
 	}
 	lua_setfield(L, -2, subnamespace);
 }
 
 /**
- * Pushes IPROTO constants related to `IPROTO_FLAG` key onto Lua stack.
+ * Pushes IPROTO constants generated from `IPROTO_FLAGS` onto Lua stack.
  */
 static void
 push_iproto_flag_constants(struct lua_State *L)
 {
-	const struct iproto_constant flags[] = {
-		{"COMMIT", IPROTO_FLAG_COMMIT},
-		{"WAIT_SYNC", IPROTO_FLAG_WAIT_SYNC},
-		{"WAIT_ACK",  IPROTO_FLAG_WAIT_ACK},
-	};
-	push_iproto_constant_subnamespace(L, "flag", flags, lengthof(flags));
+	push_iproto_constant_subnamespace(
+		L, "flag", iproto_flag_constants, iproto_flag_constants_size,
+		"IPROTO_FLAG_"
+	);
```

Comment:
Nit: Please don't put a closing parenthesis on a separate line. It makes no sense. I added a checkpatch check to prevent this.

Reply:
DOne